cron-job/templates/cronjob.yaml

```diff
@@ -20,12 +20,22 @@ spec:
           serviceAccountName: {{ include "cron-job.serviceAccountName" . }}
           containers:
           - name: {{ .Chart.Name }}
-            image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
-            imagePullPolicy: {{ .Values.image.pullPolicy }} 
+            image: "{{.Values.image.repository }}:{{ .Values.image.tag }}"
+            imagePullPolicy: {{ .Values.image.pullPolicy }}
+            {{- if .Values.execute.enable }}
+              {{- with .Values.execute.spec }}
+              {{- toYaml . | nindent 12 }}
+              {{- end }}
+            {{- end }}
             securityContext:
               runAsUser: 1000
+            {{- if .Values.persistence.volume.attach }}
+            volumeMounts:
+              - mountPath: {{.Values.persistence.volume.path }}
+                name: {{.Values.persistence.volume.name }}
+            {{- end }}
```

Comment:
this section looks like it should support more than a single volume

Reply:
it can be done, but its unlikely more then one volume will be needed, to keep it simple this is the preferred way